File: csharp/ToolGood.Algorithm/Internals/Functions/Flow/Function_ISNULLOREMPTY.cs
using System;
using System.Collections.Generic;
using ToolGood.Algorithm.Enums;

namespace ToolGood.Algorithm.Internals.Functions.Flow
{
	internal sealed class Function_ISNULLOREMPTY : Function_1
	{
		public Function_ISNULLOREMPTY(FunctionBase func1) : base(func1)
		{
		}

		public override string Name => "IsNullOrEmpty";

		public override Operand Evaluate(AlgorithmEngine engine, Func<AlgorithmEngine, string, Operand> tempParameter)
		{
			var args1 = func1.Evaluate(engine, tempParameter);
			if(args1.IsNull) { return Operand.True; }
			var textArg = ConvertToText(args1, 1);
			if(textArg.IsErrorOrNone) { return textArg; }
			return Operand.Create(string.IsNullOrEmpty(textArg.TextValue));
		}
		public override OperandType GetResultType()
		{
			return OperandType.BOOLEAN;
		}

		internal override void GetParameterTypes(NoneEngine noneEngine, List<ParameterType> result, OperandType operandType, string op = null, string val = null)
		{
			func1.GetParameterTypes(noneEngine, result, OperandType.NONE);
		}
	}

}

File: JavaScript/src/Internals/Functions/Flow/Function_ISNULLOREMPTY.js
import { Function_1 } from '../Function_1.js';
import { Operand } from '../../../Operand.js';
import { StringCache } from '../../../Internals/StringCache.js';

class Function_ISNULLOREMPTY extends Function_1 {
    get Name() {
        return "IsNullOrEmpty";
    }

    constructor(a) {
        super(a);
    }

    Evaluate(engine, tempParameter) {
        let args1 = this.a.Evaluate(engine, tempParameter);
        if (args1.IsNull) {
            return Operand.True;
        }
        let textArgs;
        if (args1.IsNotText) {
            textArgs = args1.ToText(StringCache.Function_parameter_error, "IsNullOrEmpty", 1);
            if (textArgs.IsError) { return textArgs; }
        } else {
            textArgs = args1;
        }
        return textArgs.TextValue === null || textArgs.TextValue === "" ? Operand.True : Operand.False;
    }
}

export { Function_ISNULLOREMPTY };

